youtube_dl/extractor/caffeine.py

```diff
@@ -0,0 +1,43 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+)
+
+import re
+
+
+class CaffeineIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?caffeine\.tv/.+/video/(?P<video_id>[0-9a-f-]+)'
+    _TEST = {
+        'url': 'https://www.caffeine.tv/TsuSurf/video/cffc0a00-e73f-11ec-8080-80017d29f26e',
+        'info_dict': {
+            'id': 'cffc0a00-e73f-11ec-8080-80017d29f26e',
+            'ext': 'mp4',
+            'title': 'GOOOOD MORNINNNNN #highlights',
+            'uploader': 'TsuSurf',
+            'duration': 3145,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = re.match(self._VALID_URL, url).group('video_id')
+        json_data = self._download_json('https://api.caffeine.tv/social/public/activity/' + video_id, video_id)
+        broadcast_info = json_data.get('broadcast_info')
+        title = broadcast_info['broadcast_title']
+        video_url = broadcast_info['video_url']
+
+        formats = self._extract_m3u8_formats(
+            video_url, video_id, 'mp4')
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': title,
+            'uploader': json_data['username'],
+            'duration': int_or_none(broadcast_info['content_duration']),
+            'like_count': int_or_none(json_data['like_count']),
+            'formats': formats,
+        }
```

Comment:
The optional metadata items shouldn't crash the extraction; also, the test video's JSON indicates that other standard attributes can be extracted:
```suggestion
        info = {
            'id': video_id,
            'title': title,
            'formats': formats,
        }
        info.update(traverse_obj(json_data, {
            'uploader': ((None, 'user'), 'username'),
        }, get_all=False))
        info.update(traverse_obj(json_data, {
            'like_count': ('like_count', T(int_or_none)),
            'view_count': ('view_count', T(int_or_none)),
            'comment_count': ('comment_count', T(int_or_none)),
            'tags': ('tags', Ellipsis, T(txt_or_none))),
            'is_live': 'is_live',
            'uploader': ('user', 'name'),
        }))
        info.update(traverse_obj(broadcast_info, {
            'duration': ('content_duration', T(int_or_none)),
            'timestamp': ('broadcast_start_time', T(parse_iso8601)),
            'thumbnail': ('preview_image_path', T(lambda u: urljoin(url, u))),
            'age_limit': ('content_rating', T(lambda r: r and {
                 # assume Apple Store ratings [1]
                 # 1. https://en.wikipedia.org/wiki/Mobile_software_content_rating_system
                 'FOUR_PLUS': 0,
                 'NINE_PLUS': 9,
                 'TWELVE_PLUS': 12,
                 'SEVENTEEN_PLUS': 17
             }.get(r, 17))),
        }))
        return info
```